### PR TITLE
Switch gRPC instrumentation to instrument public ServerBuilder class.

### LINKED ISSUE #1453: Fail to instrument gRPC server starting from version 1.33
Recently released [gRPC version 1.33](https://github.com/grpc/grpc-java/releases/tag/v1.33.0) broke our instrumentation point: gRPC server builders don't extend `io.grpc.internal.AbstractServerImplBuilder` that we instrument. It is fairly easy to migrate to instrumenting new superclass `ForwardingServerBuilder`, but it is hard to prevent double instrumentation.

### PR DESCRIPTION
**Pull request by anuraaga**
This would have been harder if tracing server interceptor had to be at the beginning of the list, but since it doesn't, we can just instrument the public type without problem.

Fixes #1453

### CODE REVIEW

File: instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java [hunk 1]
```diff
@@ -21,33 +28,61 @@
...
-          break;
-        }
+    public static void onEnter(
+        @Advice.This ServerBuilder<?> serverBuilder,
+        @Advice.Argument(0) ServerInterceptor interceptor) {
+      // Check against unshaded name.
+      if (interceptor
+          .getClass()
+          .getName()
+          .equals("io.opentelemetry.instrumentation.grpc.v1_5.server.TracingServerInterceptor")) {
```
Comment (trask) [inline instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java:66]: this string will get shaded too 😄

Comment (anuraaga) [reply instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java]: Oops - removed the check for now so we can come up with a better strategy globally

**Review by trask — APPROVED** [review]

File: instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java (hunk 1 above)
Comment (iNikem) [reply instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java]: So, will we have multiple copies of `TracingServerInterceptor`?

Comment (trask) [reply instrumentation/grpc-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/grpc/v1_5/GrpcServerBuilderInstrumentation.java]: See #903

**Review by iNikem — APPROVED** [review]

**Pull request merged**